site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx

```diff
@@ -57,6 +57,34 @@ export const WorkspaceBuildPageView: FC<WorkspaceBuildPageViewProps> = ({
 		defaultValue: "build",
 	});
 
+	// TODO: Use only CSS to set the height of the content.
+	// Note: On Safari, when content is rendered inside a flex container and needs
+	// to scroll, the parent container must have a height set. Achieving this may
+	// require significant refactoring of the layout components where we currently
+	// use height and min-height set to 100%.
+	// Issue: https://github.com/coder/coder/issues/9687
+	// Reference: https://stackoverflow.com/questions/43381836/height100-works-in-chrome-but-not-in-safari
+	const contentRef = useRef<HTMLDivElement>(null);
+	useLayoutEffect(() => {
+		const contentEl = contentRef.current;
+		if (!contentEl) {
+			return;
+		}
+
+		const resizeObserver = new ResizeObserver(() => {
+			const parentEl = contentEl.parentElement;
+			if (!parentEl) {
+				return;
+			}
+			contentEl.style.setProperty("height", `${parentEl.clientHeight}px`);
+		});
+		resizeObserver.observe(document.body);
+
+		return () => {
+			resizeObserver.disconnect();
+		};
+	}, []);
+
 	if (!build) {
 		return <Loader />;
 	}
```

Comment:
I think we have an edge case here that could break the effect logic:
1. Component mounts with no build
2. We set up the ref and effect to run after the first render
3. The first render completes, but because there's no build, we return out a loader and don't attach the ref to anything
4. The layout effect runs and whiffs
5. We re-render at some point with a build, but because the layout effect dependency array is empty, nothing in it re-runs
6. The only way to get the size mutations to apply at this point is if the browser resizes, but if the user never resizes anything, we're stuck with the same issue as before

Wondering if might be best to make a new component boundary and move this state there. That way, the state only mounts when we're guaranteed to have a build